InventoryCatalog/etc/di.xml

```diff
@@ -153,4 +153,10 @@
         <plugin name="adapt_verify_stock_to_negative_min_qty"
                 type="Magento\InventoryCatalog\Plugin\CatalogInventory\Model\Spi\StockStateProvider\AdaptVerifyStockToNegativeMinQtyPlugin"/>
     </type>
+    <type name="Magento\Catalog\Model\ResourceModel\Product">
+        <plugin name="delete_source_items" type="Magento\InventoryCatalog\Plugin\Catalog\Model\ResourceModel\DeleteSourceItemsPlugin"/>
+    </type>
+    <type name="Magento\Catalog\Model\Product">
+        <plugin name="process_source_items_after_product_save" type="Magento\InventoryCatalog\Plugin\Catalog\Model\Product\ProcessSourceItemsPlugin"/>
```

Comment:
I suggest to add similar plugin for InventoryLowQuantityNotification. I don't think it's cleanup automatically.

Reply:
@bubasuma, done.